optimum/habana/accelerate/accelerator.py

```diff
@@ -432,7 +432,7 @@ def _prepare_deepspeed(self, *args):
             self._prepare_one(obj, first_pass=True)
             if isinstance(obj, torch.utils.data.DataLoader)
             else convert_model(obj)
-            if isinstance(obj, torch.nn.Module) and self.state.mixed_precision == "fp8"
+            if isinstance(obj, torch.nn.Module) and self.is_fp8_enabled
```

Comment:
this definitely breaks other things 😅 notice that the mixed_precision that was passed during accelerator and accelerator state initialization is not used at all ..